File: 프로그래머스/0/120888. 중복된 문자 제거/중복된 문자 제거.java
class Solution {
    public String solution(String my_string) {
        String str = my_string;

        StringBuilder sb = new StringBuilder();
        boolean[] seen = new boolean[256]; 

        for (int i = 0; i < str.length(); i++) {
            char c = str.charAt(i);
            if (!seen[c]) {
                seen[c] = true;
                sb.append(c);
            }
        }

        return sb.toString();

    }
}

File: 프로그래머스/0/120888. 중복된 문자 제거/중복된 문자 제거.js
function solution(my_string) {
    let str = my_string;

    let seen = new Array(256).fill(false);
    let result = [];

    for (let i = 0; i < str.length; i++) {
        let c = str.charCodeAt(i);
        if (!seen[c]) {
            seen[c] = true;
            result.push(str.charAt(i));
        }
    }

    return result.join('');
}